metaflow/plugins/aws/aws_client.py

```diff
@@ -45,12 +47,26 @@ def get_client(module, with_error=False, params={}):
             return boto3.session.Session(**cached_aws_sandbox_creds).client(
                 module, **params
             )
+        session = boto3.session.Session()
+        if s3_role_arn:
```

Comment:
Minor nit - we can rename this from `s3_role_arn` to just `role_arn` - there isn't anything s3 specific here.